specifications/13-constraint-operators.json

```diff
@@ -0,0 +1,729 @@
+                        "constraints": [
+                            {
+                                "contextName": "someValue",
+                                "operator": "NUM_GTE",
+                                "value": 12
+                            }
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "F5.numLT",
+                "description": "contains",
+                "enabled": true,
+                "strategies": [
+                    {
+                        "name": "default",
+                        "parameters": {},
+                        "constraints": [
+                            {
+                                "contextName": "someValue",
+                                "operator": "NUM_LT",
+                                "value": 12
+                            }
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "F5.numLTE",
+                "description": "contains",
+                "enabled": true,
+                "strategies": [
+                    {
+                        "name": "default",
+                        "parameters": {},
+                        "constraints": [
+                            {
+                                "contextName": "someValue",
+                                "operator": "NUM_LTE",
+                                "value": 12
+                            }
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "F6.number-range",
+                "description": "range of numbers",
+                "enabled": true,
+                "strategies": [
+                    {
+                        "name": "default",
+                        "parameters": {},
+                        "constraints": [
+                            {
+                                "contextName": "someValue",
+                                "operator": "NUM_GT",
+                                "value": 12
+                            },
+                            {
+                                "contextName": "someValue",
+                                "operator": "NUM_LT",
+                                "value": 16
+                            }
+                        ]
+                    }
+                ]
+            },
+            {
+                "name": "F7.dateAfter",
+                "description": "dates",
+                "enabled": true,
+                "strategies": [
+                    {
+                        "name": "default",
+                        "parameters": {},
+                        "constraints": [
+                            {
+                                "contextName": "currentTime",
+                                "operator": "DATE_AFTER",
+                                "value": "2022-01-29T13:00:00.000Z"
```

Comment:
This is probably a minor thing but!  Should `DATE_BEFORE` and `DATE_AFTER` be inclusive (i.e. LT versus LTE)? 

I don't see this being a huge issue (there's a vanishingly small chance people will notice) but figured I'd ask as long as I was writing unit tests.  In the Python client, it's currently coded to be inclusive.

Reply:
The meaning of the word "AFTER" seems to be closer to GT. 

For this reason I think we should require it to be exclusive.

Reply:
* How about also allowing using `DATE_GTE`, `DATE_GT` as more precise definitions of `DATE_AFTER`?
  (and the equivalent for `_LTE`, `_LE` of for `DATE_BEFORE`)

* It would have been useful to also have `SEMVER_GTE` and `_LTE`

* It would also nice to have tests for other strings which are definitely not SEMVER, so we have predictable behaviour too. Maybe just do in the current locale string ordering when one of the values is not a SEMVER? (optionally log a warning?) Like :
  * "v1.0.0" <=> "1.0.0"  ( I know that [versions prefixed with `v` are not SEMVER](https://semver.org/#is-v123-a-semantic-version) but they are still widely use, so it would be good that we at least test for them)
  * "2.0.0-beta.1" <=> "2.0.0b1"
  * "abc" <=> "abz"
  * "1.0" <=> "abc"

I understand that this feedback might be a bit late, as the spec seems to be closed now, but I thought it was still worth giving my 2 cents.